src/db.ts

```diff
@@ -1,27 +1,204 @@
-        console.log("successfully connected to DB");
-    } finally {
+    }
+
+    private async disconnect(): Promise<void> {
         await mongoose.disconnect();
     }
+
+    public async createRoom(roomData: RoomSchema): Promise<void> {
+        try {
+            const roomModel = getModelForClass(RoomSchema);
+            const room = await roomModel.create(roomData);
+            await room.save();
+            console.log(`New room created with id (handler): ${room._id}`);
+        } catch (error) {
+            console.error("Room create error (handler):", error);
+        }
+    }
+
+    public async updateRoom(roomId: string, updateData: {
+        RoomState?: string,
+        PlayerIds?: Array<string>,
+        HostWsId?: string
+    }): Promise<void> {
+        const roomModel = getModelForClass(RoomSchema);  
+        try {
+            const res = await roomModel.updateOne({ _id: roomId }, updateData);
+            console.log(`Room updated with id (handler): ${roomId}`);
+        } catch (error) {
+            console.error("Room update error (handler):", error);
+        }
+    }
+
+    public async createQuiz(quizData: QuizSchema): Promise<void> {
+        try {
+            const quizModel = getModelForClass(QuizSchema);
+            const quiz = await quizModel.create(quizData);
+            await quiz.save();
+            console.log(`New quiz created with id (handler): ${quiz._id}`);
+        } catch (error) {
+            console.error("Quiz create error (handler):", error);
+        }
+    }
+
+    public async updateQuizList(quizId: string, updateData: {
+        Questions?: Array<string>,
+    }): Promise<void> {
+        const quizModel = getModelForClass(QuizSchema);
+        try {
+            const res = await quizModel.updateOne({ _id: quizId }, updateData);
+            console.log(`Quiz updated with id (handler): ${quizId}`);
+        } catch (error) {
+            console.error("Quiz update error (handler):", error);
+        }
+    }
+
+    public async createQuestionHandler(questionData: QuestionSchema): Promise<void> {
+        try {
+            const questionModel = getModelForClass(QuestionSchema);
+            const question = await questionModel.create(questionData);
+            await question.save();
+            console.log(`New question created with id (handler): ${question._id}`);
+        } catch (error) {
+            console.error("Question create error (handler):", error);
+        }
+    }
+
+    public async updateQuestionHandler(questionId: string, updateData: {
+        Question?: string,
+        PossibleAnswers?: Map<string, string>,
+        CorrectAnswer?: string,
+        QuestionType?: string,
+    }): Promise<void> {
+        const questionModel = getModelForClass(QuestionSchema);
+        try {
+            const res = await questionModel.updateOne({ _id: questionId }, updateData);
+            console.log(`Question updated with id (handler): ${questionId}`);
```

Comment:
specifying the difference between handler and interface is meaningless. These log/error messages should be helpful for debugging and validating behavior. ie `successful insert document <x> with roomid <y> into <db>` or `failed to update roomId <x> with new state <y> in db <z>, <actual error message>`

Reply:
applies to all